scheduler/src/test/java/org/cloudfoundry/autoscaler/scheduler/rest/ScheduleRestControllerTest.java

```diff
@@ -282,6 +291,21 @@ public void testCreateSchedule_without_appId() throws Exception {
 		resultActions.andExpect(status().isNotFound());
 
 	}
+	
+	@Test
+	public void testCreateSchedule_without_guid() throws Exception {
+
+		String appId = "appId";
+		ObjectMapper mapper = new ObjectMapper();
+		ApplicationSchedules applicationPolicy = TestDataSetupHelper.generateApplicationPolicy(1, 0);
+		String content = mapper.writeValueAsString(applicationPolicy);
+
+		ResultActions resultActions = mockMvc
+				.perform(put("/v2/schedules/" + appId).contentType(MediaType.APPLICATION_JSON).content(content));
+
+		resultActions.andExpect(status().isInternalServerError());
```

Comment:
Just notify, you can add param argument into @RequestMapping like below code. I haven't verified however it might help to solve this situation.

```
@RequestMapping(method = RequestMethod.PUT, params={"guid"})
```